src/ProtocolStreamReader.php

```diff
@@ -54,6 +54,9 @@ public function __construct($input)
                             $this->parsingMode = ParsingMode::HEADERS;
                             $this->headers = [];
                             $this->buffer = '';
+
+                            // after reading a full message, leave to allow different tasks to run
+                            return;
```

Comment:
Hm - is there no way to only read until the buffer is empty? I thought you said when using a socket it would work non-blocking, so it would return an empty string as soon as the buffer is empty?